gsrs-rest-api/src/main/java/gsrs/api/GsrsEntityRestTemplate.java

```diff
@@ -67,7 +67,7 @@ public long count() throws IOException {
 
     public <S extends T> Optional<S> findByResolvedId(String anyKindOfId) throws IOException{
         ResponseEntity<String> response = doGet("("+anyKindOfId + ")",String.class);
-        if(response.getStatusCodeValue() == 404) {
+        if(!response.getStatusCode().is2xxSuccessful()) {
```

Comment:
This is probably okay, but I do wonder what happens on 302/etc. I'm not sure whether redirects are auto followed, but they should be anyway. Worth confirming that the RestTemplate is set to follow redirects.